### ISSUE TITLE
Stop downloading without deleting file

### ISSUE THREAD

**Issue by ihor-zinchenko** [hgouveia/node-downloader-helper#22]
How I can stop downloading without file deleting?

I have electron app with node-downloader-helper
Before closing the application, I've got an error http://joxi.ru/brRap3JI7dLdBA, I think that may be related to downloading the file. I do pause before closing the application.

**Comment by hgouveia** [comment]
> How I can stop downloading without file deleting?
> 
> I have electron app with node-downloader-helper
> Before closing the application, I've got an error http://joxi.ru/brRap3JI7dLdBA, I think that may be related to downloading the file. I do pause before closing the application.

Yeap this lib, remove the file automatically when **Stop** is called or fails, but if you call **Pause**  it does the same as **Stop** but without the file removing,  you mentioned that you pause on close, so it shouldn't remove your file, what I think is happening, is that the close is not finished after close, and this probably is a bug of my part, 

I will investigate and fix it in the new release, in the meanwhile, I suggest to do is just a setTimeout after pause all of them with some milliseconds, like 1000, (1sec), and then trigger Close on electron

**Comment by IhorVimmi** [comment]
still happens but not always. if I pause and unpause downloading I've got error always but when I start and pause once before close sometimes I didn't get error

**Comment by NickPepper** [comment]
I got exactly the same issue in my Electron-app.

**Comment by hgouveia** [comment]
@IhorVimmi @NickPepper  , 

I just added a fix for this on my dev branch, could you test it? using `npm install --save hgouveia/node-downloader-helper#dev`

when using  `stop` now will wait until the file is closed, and I added a new option called `removeOnStop` that you can set to `false`  to avoid the file deletion

**Comment by IhorVimmi** [comment]
@hgouveia, unfortunately, the same, I used stop instead of pause but I got the same result

**Comment by hgouveia** [comment]
@IhorVimmi is possible you can setup a minimal electron app, reproducing your issue and send it to me?

**Comment by IhorVimmi** [comment]
@hgouveia sure, I'll prepare and send it tomorrow, thank you for your work!

**Comment by hgouveia** [comment]
> @hgouveia sure, I'll prepare and send it tomorrow, thank you for your work!

hello @IhorVimmi any luck with this?

**Comment by hgouveia** [comment]
> > @hgouveia sure, I'll prepare and send it tomorrow, thank you for your work!
> 
> hello @IhorVimmi any luck with this?

@IhorVimmi any luck with this?

**Comment by davidchalifoux** [comment]
Looking forward to this being on the main branch. I'm using this in a CLI tool and I am calling `.stop()` when the process is killed, but the file is not being deleted.

**Comment by hgouveia** [comment]
> Looking forward to this being on the main branch. I'm using this in a CLI tool and I am calling `.stop()` when the process is killed, but the file is not being deleted.

Hello @davidchalifoux , this feature was already published on 1.0.12 , where you have two new options `removeOnStop` and `removeOnFail` both set as `true` by default, but keep in mind that `stop` returns a promise,  so you will need to wait until finishes to close, could you do something like this

```
process.on('SIGTERM',() =>{
   // finally available since node10
   dl.stop().finally( () => process.exit() );
});
```

**Comment by davidchalifoux** [comment]
> > Looking forward to this being on the main branch. I'm using this in a CLI tool and I am calling `.stop()` when the process is killed, but the file is not being deleted.
> 
> Hello @davidchalifoux , this feature was already published on 1.0.12 , where you have two new options `removeOnStop` and `removeOnFail` both set as `true` by default, but keep in mind that `stop` returns a promise, so you will need to wait until finishes to close, could you do something like this
> 
> ```
> process.on('SIGTERM',() =>{
>    // finally available since node10
>    dl.stop().finally( () => process.exit() );
> });
> ```

This worked, thanks again!
Totally forgot to try that.

**Comment by hgouveia** [comment]
closing this, fixed on `1.0.12`
